server/models/dataset.py

```diff
@@ -327,16 +296,49 @@ def reviewImages(self, dataset, acceptedImages, flaggedImages, user):
                 Folder.countFolders(prereviewFolder)) == 0:
             Folder.remove(prereviewFolder)
 
-    def registerMetadata(self, dataset, csvFile):
+    def registerMetadata(self, dataset, csvFile, user, sendMail=False):
         """Register a .csv file containing metadata about images."""
-        # Add image metadata information to list
-        registeredMetadata = dataset['meta'].get('metadata', [])
-        registeredMetadata.append({
+        # Check if image metadata is already registered
+        if self.findOne({'meta.metadataFiles.fileId': csvFile['_id']}):
+            raise ValidationException('Metadata file is already registered.')
+
+        # Add image metadata file information to list
+        now = datetime.datetime.utcnow()
+        metadataFiles = dataset['meta'].get('metadataFiles', [])
+        metadataFiles.append({
             'fileId': csvFile['_id'],
-            'time': datetime.datetime.utcnow()
+            'user': user['_id'],
+            'time': now
         })
         dataset = self.setMetadata(dataset, {
-            'metadata': registeredMetadata
+            'metadataFiles': metadataFiles
         })
 
+        # Send email notification
+        if sendMail:
+            host = mail_utils.getEmailUrlPrefix()
+            self._mailGroup(
+                groupName='Dataset QC Reviewers',
+                templateFilename='registerMetadataNotification.mako',
+                templateParams={
+                    'host': host,
+                    'dataset': dataset,
+                    'user': user,
+                    'csvFile': csvFile,
+                    'date': now.replace(microsecond=0)
+                },
+                subject='ISIC Archive: Dataset Metadata Notification')
+
         return dataset
+
+    def _mailGroup(self, groupName, templateFilename, templateParams, subject):
+        """Send email to all members of a group."""
+        Group = self.model('group')
+
+        group = Group.findOne({'name': groupName})
+        if not group:
+            raise GirderException('Could not load group: %s.' % groupName)
+        emails = [member['email'] for member in Group.listMembers(group)]
+        if emails:
+            html = mail_utils.renderTemplate(templateFilename, templateParams)
+            mail_utils.sendEmail(to=emails, subject=subject, text=html)
```

Comment:
I'm debating whether we should loop over members here and send single emails directly to users, to avoid compromising the privacy of our users' email addresses.

Since the groups we're sending emails to now are internal administrators, for now, I think we should actually keep it as one email with multiple recipients, as it could be convenient for a recipient to Reply-All and notify other group members of something. If we start sending mails to "external" user group, like the entirety of `Dataset Contributors`, then we should consider the looping approach, or maybe add a function parameter to control single vs mass emails.